src/libraries/Microsoft.PowerFx.Core/Functions/IAsyncTexlFunction5.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core.Functions
+{    
+    // Texl function interface with IServiceProvider
+    internal interface IAsyncTexlFunction5
```

Comment:
>IAsyncTexlFunction5 [](http://example.com/codeflow?start=23&length=19)

Is there a good way to squish these `IAsyncTexlFunction` into less numbers of interfaces?

Reply:
We'd like to - but that's a separate refactoring

Reply:
Reopening this comment.

I understand that this is a separate work item, but we can at least stop the bleeding ("when you find yourself in a hole, the first thing to do is to stop digging"). IAsyncTexlFunction4 takes the same arguments, except the first one is a TimeZoneInfo. Since it's an internal interface, we can change it to take an IServiceProvider instead, and retrieve the TimeZoneInfo from the provider (and update our implementations of that interface)

Reply:
As part of the refactoring, we'll get rid of all other IAsyncTexlFunctionX

Reply:
fortunately, these are internal and so we have an option to fix.